src/main/java/com/aws/greengrass/mqtt/bridge/util/TopicMatcher.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.mqtt.bridge.util;
+
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Implementation adapted from <a href="https://github.com/eclipse/paho.mqtt.java/blob/f4e0db802a4433645ef011e711646a09ec9fae89/org.eclipse.paho.client.mqttv3/src/main/java/org/eclipse/paho/client/mqttv3/MqttTopic.java#L253-L321">paho</a>.
+ */
+@SuppressWarnings("PMD.UseVarargs")
+public final class TopicMatcher {
+    private static final String TOPIC_LEVEL_SEPARATOR = "/";
```

Comment:
We have this in Nucleus MqttTopic.

Reply:
private access tho

Reply:
I mean the entire functionality, not the constant.  Any reason why you cannot use MqttTopic class the same way you use this class?

Reply:
🙃  ah right, yeah that'd work, will replace